src/clims/api/serializers/models/substance.py

```diff
@@ -1,47 +1,17 @@
 from __future__ import absolute_import
 
 from clims.models.substance import Substance
-from rest_framework import serializers
-from rest_framework.fields import Field
-
-
-class PropertyValueField(Field):
-    def to_representation(self, obj):
-        return obj
-
-    def to_internal_value(self, data):
-        return data
-
-
-class PropertyTypeSerializer(serializers.Serializer):
-    id = serializers.IntegerField(read_only=True)
-
+from clims.api.serializers.models.extensible_property import ExtensiblePropertySerializer
 
-class PropertyOverviewSerializer(serializers.Serializer):
-    id = serializers.IntegerField(read_only=True)
-    type = PropertyTypeSerializer()
-    value = PropertyValueField()
-
-
-class PropertiesField(Field):
-    def to_representation(self, obj):
-        return {key: prop.value for key, prop in obj.items()}
-
-    def to_internal_value(self, data):
-        # TODO
-        return data
-
-
-class VersionField(Field):
-    def to_representation(self, obj):
-        return obj
+from rest_framework import serializers
+from rest_framework.fields import DictField
 
 
 class SubstanceSerializer(serializers.Serializer):
-    id = serializers.CharField(read_only=True)
+    id = serializers.IntegerField(read_only=True)
```

Comment:
This was an interesting thing. I decided to use a string, as it's an implementation detail that it actually is an integer in the backend. Since the API user will never have to treat the ID as anything but a string (they are for example not going to add two IDs), I felt that it should be a string. At the moment however, it really doesn't matter and we will probably never want to use anything else anyways (e.g. GUIDs). So perhaps it's right to just KISS it.